packages/rich-text/src/plugins/EmbeddedEntityInline/index.tsx

```diff
@@ -94,24 +99,40 @@ interface ToolbarEmbeddedEntityInlineButtonProps {
   isButton?: boolean;
 }
 
-async function selectEntityAndInsert(editor, sdk: FieldExtensionSDK) {
+async function selectEntityAndInsert(
+  editor,
+  sdk: FieldExtensionSDK,
+  logAction: TrackingPluginActions['onShortcutAction'] | TrackingPluginActions['onToolbarAction']
+) {
+  logAction('openCreateEmbedDialog', { nodeType: INLINES.EMBEDDED_ENTRY });
+
   const config = {
     ...newEntitySelectorConfigFromRichTextField(sdk.field, INLINES.EMBEDDED_ENTRY),
     withCreate: true,
   };
   const selection = editor.selection;
 
-  const entry = await sdk.dialogs.selectSingleEntry<Entry>(config);
-  focus(editor); // Dialog steals focus from editor, return it.
-  if (!entry) return;
-
-  const inlineEntryNode = createInlineEntryNode(entry.sys.id);
-
-  // Got to wait until focus is really back on the editor or setSelection() won't work.
-  setTimeout(() => {
-    Transforms.setSelection(editor, selection);
-    Transforms.insertNodes(editor, inlineEntryNode);
-  }, 0);
+  try {
+    const entry = await sdk.dialogs.selectSingleEntry<Entry>(config);
+    focus(editor); // Dialog steals focus from editor, return it.
+    if (!entry) return;
+
+    const inlineEntryNode = createInlineEntryNode(entry.sys.id);
+
+    // Got to wait until focus is really back on the editor or setSelection() won't work.
+    setTimeout(() => {
+      Transforms.setSelection(editor, selection);
+      Transforms.insertNodes(editor, inlineEntryNode);
+    }, 0);
+
+    logAction('insert', { nodeType: INLINES.EMBEDDED_ENTRY });
+  } catch (error) {
+    if (error) {
+      throw error;
+    } else {
+      logAction('cancelCreateEmbedDialog', { nodeType: INLINES.EMBEDDED_ENTRY });
+    }
+  }
```

Comment:
How does this part work? What do we get when `sdk.dialogs.selectSingleEntry()` rejects?

Reply:
Based on the [docs](https://www.contentful.com/developers/docs/extensibility/app-framework/sdk/#dialogs), looks like it doesn't reject at all. Instead, it returns `null`

That means the logging of `cancelCreateEmbedDialog` should be done in the `if(!entry) return` above in line 118

Reply:
This method throws an error in reality. I guess the docs are wrong or we have a different version of the SDK.